core/src/main/java/org/apache/iceberg/IncrementalDataTableScan.java

```diff
@@ -74,7 +74,7 @@ public TableScan appendsAfter(long newFromSnapshotId) {
     Set<Long> snapshotIds = Sets.newHashSet(Iterables.transform(snapshots, Snapshot::snapshotId));
     Set<ManifestFile> manifests = FluentIterable
         .from(snapshots)
-        .transformAndConcat(s -> s.dataManifests())
+        .transformAndConcat(Snapshot::dataManifests)
```

Comment:
Forgive me if this is a silly question, as I am more of a scala developer traditionally and it's been a long day.

Does this change the semantics? Previously it was calling the `dataManifests` method of each value in the iterable. And I notice later on there's a call to `this::isNamespace`. Is this method reference expected to be called in a static fashion (the interface doesn't declare it static) or will it be called on each instance of the iterable? I only ask because `BaseSnapshot#dataManifests` is a potentially side-effecting function. 

But again, I might just be working too late with too much Scala in the brain, so forgive me if I've just not brushed up on my Java lately. My brain is _really_ trying to `flatMap` here so I could very well just need to brush up on my Java.

Reply:
This is confusing. The this::reference is a call to the method within the instance of the host class, not the instance within the stream.

For example
```java
class Actions {
 void bark( Dog dog) {}
 void barks ( List<Dog> dogs){
  Dogs.stream.foreach(this::bark); // x -> this.bark(x)
}
```

Vs using the class name which applies the method of the stream's instance to itself.

```java
class Actions{
  class Dog {
    void bark() {}
  }
 void barks ( List<Dog> dogs){
   Dogs.stream.foreach(Dog::bark); // Dog x -> x.bark()
}
```

https://docs.oracle.com/javase/tutorial/java/javaOO/methodreferences.html

Reply:
Thanks @RussellSpitzer! I probably overly convoluted my question by bringing `this` into it, but your answer succinctly helped me understand / relearn some of the more java-y parts that I might not have used in a bit. As always, greatly appreciate you taking the time to help me learn!